content/develop/concepts/architecture/fragments

```diff
@@ -0,0 +1,143 @@
+---
+title: Working withwq fragments and partial reruns
+slug: /develop/concepts/architecture/fragments
+---
+
+# Working with fragments and partial reruns
+
+Reruns are a central part of every Streamlit app. When users interact with widgets, your script reruns from top to bottom and your app's frontend is updated. Streamlit provides several features to help you develop your app within this execution model. Fragments were introduced in Streamlit version 1.33.0 to allow rerunning a portion or your code instead of your full script. As your app grows larger and more complex, partial reruns help your app efficient and performant.
```

Comment:
at the end:

> fragments can make your app more efficient and performant

Maybe also mention something about control flow / app logic being easier to reason about (?)

Reply:
"Fragments give you finer control over your app's execution flow."